app/controllers/auth_controller.rb

```diff
@@ -1,39 +1,92 @@
   # GETs should be safe (see http://www.w3.org/2001/tag/doc/whenToUseGet.html)
-  verify :method => :post, :only => [ :login, :logout ],
+  verify :method => :post, :only => [ :login, :relogin, :logout ],
     :redirect_to => { :action => :list }
 
   def action_allowed?
     case params[:action]
-    when 'login', 'logout', 'login_failed', 'google_login'
+    when 'login', 'logout', 'relogin', 'login_failed', 'google_login'
       true
     else
       current_role_name.eql?("Super-Administrator")
     end
   end
 
+   def relogin
+    if request.get?
+      AuthController.clear_session(session)
+    else
+       user = User.find_by_login(params[:login][:name])
+       #aise "error"
+    if(user.next_login_time<=DateTime.now)
+       if user and user.valid_password?(params[:login][:password]) && simple_captcha_valid?
+        user.login_attempts=0
+        user.save
+        after_login(user)
+       else
+        failed_authentication(user)
+       end
+      else
+        flash[:error] = "Wait till #{user.next_login_time} for next login attempt"
+        redirect_to :controller => 'content_pages', :action => 'view'
+      end
+    end
+  end  #def relogin
+
   def login
     if request.get?
       AuthController.clear_session(session)
     else
-      user = User.find_by_login(params[:login][:name])
+       user = User.find_by_login(params[:login][:name])
+       #aise "error"
+    if(user.next_login_time<=DateTime.now)
+      if (user.login_attempts >= 3)
+          AuthController.clear_session(session)
+          render 'content_pages/relogin'
+      else
        if user and user.valid_password?(params[:login][:password])
-         after_login(user)
+          user.login_attempts=0
+          user.save
+          after_login(user)
        else
-         logger.warn "Failed login attempt"
-         flash[:error] = "Incorrect Name/Password"
-         redirect_to :controller => 'password_retrieval', :action => 'forgotten'
+        failed_authentication(user)
        end
+      end
+      else
+        flash[:error] = "Wait till #{user.next_login_time} for next login attempt"
+        redirect_to :controller => 'content_pages', :action => 'view'
+      end
     end
   end  #def login
 
+  def failed_authentication(user)
+    if(user.login_attempts < 3)
+     user.login_attempts=user.login_attempts+1
+     user.save
+     logger.warn "Failed login attempt"
+     flash[:error] = "Incorrect Name/Password"
+     redirect_to :controller => 'content_pages', :action => 'view'
+    else
+      exponential_backoff(user)
+    end
+  end 
+
+  def exponential_backoff(user)
+    # function to handle exponential backoff afterfailedlogin attempt
+    user.login_attempts=user.login_attempts+1
+    interval=2**(user.login_attempts-3)
+    user.next_login_time=DateTime.now+interval.minutes
+    user.save
+    logger.warn "Failed login attempt: Account Blocked"
+    flash[:error] = "Account is Blocked for #{interval} minutes"
+    render 'content_pages/relogin'
```

Comment:
Is this possible that the user change the url from relogin to login after the 1st try fails?